library/cwm/examples/object_api.rb

```diff
@@ -0,0 +1,78 @@
+# Simple example to demonstrate object API for CWM
+
+require_relative "example_helper"
+
+require "cwm/widget"
+
+Yast.import "CWM"
+Yast.import "Wizard"
+
+class LuckyNumberWidget < CWM::IntField
+  attr_reader :result, :minimum, :maximum
+
+  def initialize
+    @minimum = 0
+    @maximum = 1000
+    self.widget_id = "lucky_number"
+  end
+
+  def label
+    _("Lucky number")
+  end
+
+  def store(_widget, _event)
+    @result = value
+  end
+end
+
+class GenerateButton < CWM::PushButtonWidget
+  def initialize(lucky_number)
+    self.widget_id = "generate"
+    @lucky_number = lucky_number
+  end
+
+  def label
+    _("Generate Lucky Number")
+  end
+
+  def handle(widget, event)
+    return unless my_event?(widget, event)
```

Comment:
This should really be the default behavior of AbstractWidget. If you think that some widgets will need to process other events then that can be enabled in their constructor.